### PR TITLE
Stats - Add the ability to mark a referrer as spam

### LINKED ISSUE #2232: Add the ability to mark a site as Spam under “Referrers”
Title has it.

### PR DESCRIPTION
**Pull request by daniloercoli**
Fix #2232 

cc @astralbodies

### CODE REVIEW

**Comment by daniloercoli** [comment]
Note to the reviewer: ping me so I can give the command line shortcuts to restore the spam state for the testing site.

**Comment by astralbodies** [comment]
Cool - its easier to mark as spam on the API side than I thought it would be!

**Comment by tonyr59h** [comment]
When I mark a referrer as spam on the web and refresh the app stats it doesn't see the change. Expected?

File: WordPress/src/main/java/org/wordpress/android/ui/stats/ReferrerSpamHelper.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    // return the domain of the passed ReferrerGroupModel or null.
+    private static String getDomain(ReferrerGroupModel group) {
+        // Use the URL value given in the JSON response, or use the groupID that doesn't contain the schema.
+        final String spamDomain = group.getUrl() != null ? group.getUrl() : "http://" + group.getGroupId();
+        return UrlUtils.isValidUrlAndHostNotNull(spamDomain) ? UrlUtils.getDomainFromUrl(spamDomain) : null;
+    }
+
+    public static boolean isSpamActionAvailable(ReferrerGroupModel group) {
+        String domain = getDomain(group);
+        return !TextUtils.isEmpty(domain) && !domain.equals("wordpress.com");
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/ReferrerSpamHelper.java]: Are jetpack sites supported?

File: WordPress/src/main/java/org/wordpress/android/ui/stats/ReferrerSpamHelper.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                        volleyError.getMessage(), volleyError);
+            }
+            if (mActivityRef.get() == null || mActivityRef.get().isFinishing()) {
+                return;
+            }
+
+            mReferrerGroup.isRestCallInProgress = false;
+            ToastUtils.showToast(mActivityRef.get(),
+                    mActivityRef.get().getString(R.string.stats_referrers_spam_generic_error),
+                    ToastUtils.Duration.LONG);
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/ReferrerSpamHelper.java]: Might be my bias against toasts in general, but I think a SHORT duration would be fine. cc @mattmiklic

**Comment by tonyr59h** [comment]
Looks good! I'll wait for a response to the questions and see what Matt says about the toast length before merging.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/ReferrerSpamHelper.java (hunk 2 above)
Comment (mattmiklic) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/ReferrerSpamHelper.java]: Given the length of the message, I'd say long, to be sure people have time to notice the message and read it before it disappears.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/ReferrerSpamHelper.java (hunk 1 above)
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/ReferrerSpamHelper.java]: It works on all sites, as long as the target != wordpress.com.

**Comment by daniloercoli** [comment]
> When I mark a referrer as spam on the web and refresh the app stats it doesn't see the change. Expected?

Yes, the app has a cache of 10 minutes. We can build something cool by sending a hidden PN to all devices of the user, that will reset stats in background. But I think we can live with it for now :)

**Comment by tonyr59h** [comment]
> But I think we can live with it for now

:+1: 
:shipit:

**Pull request merged**